app/assets/javascripts/payment.js

```diff
@@ -0,0 +1,30 @@
+jQuery(function($) {
+  $('#payment-form').submit(function(event) {
+    var $form = $(this);
+
+    // Disable the submit button to prevent repeated clicks
+    $form.find('button').prop('disabled', true);
+
+    Stripe.card.createToken($form, stripeResponseHandler);
+
+    // Prevent the form from submitting with the default action
+    return false;
+  });
+  
+  function stripeResponseHandler(status, response) {
+    var $form = $('#payment-form');
+
+    if (response.error) {
+      // Show the errors on the form
+      $form.find('.payment-errors').html('<div class="alert alert-danger">' + response.error.message);
```

Comment:
Good spot, I'll change that now although I just checked the view and it seems to close it automatically. Not sure how it does this this or if it's a browser specific thing but for clarity I should close it.  Getting to used to writing haml and not needing to close tags!